cspell.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
Do we need both the json and yaml file in the root?

Reply:
The JSON file is required today because of how `check-spelling-in-changed-files.ps1` works (it builds a list of files and puts it in the root `cspell.json` file and doesn't read YAML. 

The `cspell.yaml` file is more user friendly so I migrated us to that format